crane/container.go

```diff
@@ -639,16 +639,36 @@ func (c *container) Logs(follow bool, tail string) (stdout, stderr io.Reader) {
 }
 
 // Push container
-func (c *container) Push() {
-	if len(c.Image()) > 0 {
-		fmt.Printf("Pushing image %s ... ", c.Image())
-		args := []string{"push", c.Image()}
+func (c *container) Push(tag string) {
+	imageName := c.nameWithTagOverride(tag)
+	if len(imageName) > 0 {
+		fmt.Printf("Pushing image %s ... ", imageName)
+		args := []string{"push", imageName}
 		executeCommand("docker", args)
 	} else {
 		print.Noticef("Skipping %s as it does not have an image name.\n", c.Name())
 	}
 }
 
+func (c *container) nameWithTagOverride(tag string) string {
+	imageName := c.Image()
+	fmt.Println(imageName)
```

Comment:
I don't think this should be printed here. Also below. I don't think anything should be printed in this method.